lib/npg_pipeline/function/stage2pp.pm

```diff
@@ -72,8 +94,10 @@ sub create {
     }
   }
 
+  @definitions = grep { $_ } @definitions;
+
   if (@definitions) {
-    (@definitions == @{$self->_output_dirs}) or $self->logcroak(
+    (@definitions <= @{$self->_output_dirs}) or $self->logcroak(
       sprintf 'Number of definitions %i and output directories %i do not match',
```

Comment:
I think this error message needs to be updated to match the condition checked. 'Less or equal than' vs 'match'.